pkg/hostagent/api/client/client.go

```diff
@@ -62,3 +65,35 @@ func (c *client) Info(ctx context.Context) (*api.Info, error) {
 	}
 	return &info, nil
 }
+
+func (c *client) DriverConfig(ctx context.Context, config interface{}) (interface{}, error) {
+	u := fmt.Sprintf("http://%s/%s/driver/config", c.dummyHost, c.version)
+	method := "GET"
+	var body io.Reader
+	if config != nil {
+		method = "PATCH"
+		b, err := json.Marshal(config)
+		if err != nil {
+			return nil, err
+		}
+		body = bytes.NewBuffer(b)
+	}
```

Comment:
Doing both GET and PATCH in the same function is confusing, both for understanding this code, and both for code using this function. If we want to avoid duplicate code for sending a request and getting a response, move the common code to a helper.